### PR TITLE
[TypeUtils.findOverridenMethod] Skip found methods with default access and not same package

### LINKED ISSUE #3599: `TypeUtils.findOverriddenMethod` misidentifies package scoped method from class in different package
Greetings. 

We've found a problem in the [MissingOverrideAnnotation ](org.openrewrite.staticanalysis.MissingOverrideAnnotation)recipe, by which in very specific circunstances, a method will be wrongly annotated, generating compilation issues in processed projects. 

The error can be observed whenever:

- A class extends/implements from a supertype.
- The supertype contains a method which is not exposed externally (that is, with package/default visibility)
- The extending class declares a new method with the same signature.
- The recipe will wrongly consider this copycat method an override and assign the annotation, resulting in compilation failures.

## What version of OpenRewrite are you using?
staticAnalysis 1.0.8
RecipeBom 2.3.0
RewriteMavenPlugin 5.7.1

## How are you running OpenRewrite?

Through maven plugin in multimodule projects.

## What is the smallest, simplest way to reproduce the problem?

Given the following example superclass:

```java
package org.example;

public class A {

    public void accessible() {
       // Open for overriding
    }

    void notAccessible() {
        // This method is not visible from subclass due to default access and different package.
    }
}
```

And the following extending class:

```java
package org.example.impl;

import org.example.A;

class B extends A {

    public void accessible() {
        super.accessible();
    }

    void notAccessible() {
        // Same signature than super method but not an override.
    }
}
```

## What did you expect to see?
The recipe should add the override annotation to the accesible method.

```java
package org.example.impl;

import org.example.A;

class B extends A {

    @Override
    public void accessible() {
        super.accessible();
    }

    void notAccessible() {
        // Same signature than super method but not an override.
    }

}
```

## What did you see instead?

The notAccessible() method is considered an override and is annotated.
This results in compilation failure as notAccesible() can't override the supertype.

```java
package org.example.impl;

import org.example.A;

class B extends A {

    @Override
    public void accessible() {
        super.accessible();
    }

    @Override
    void notAccessible() {
        // Same signature than super method but not an override.
    }

}
```

## Are you interested in [contributing a fix to OpenRewrite](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes)?
Yes, we can contribute a fix to skip override annotations for these specific cases.

### PR DESCRIPTION
**Pull request by joteri**
## What's your motivation?

Fixes #3599 

## Anything in particular you'd like reviewers to focus on?

Since it is an utility method that is invoked from different scenarios check if this change can affect any other behaviors.

## Anyone you would like to review specifically?

@timtebeek

### Checklist
- [ ] I've added unit tests to cover both positive and negative cases
- It seems affected method is being invoked from another unit tests already so specific test was not added.
- Should add a specific unit test for `MissingOverrideAnnotation` recipe to verify this bugfix.
- [ ] I've added the license header to any new files through `./gradlew licenseFormat`
- no new files were added
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

**Comment by knutwannheden** [comment]
It would be very nice if we could get a unit test for this. Something like this:

```java
    @Test
    void packagePrivateMethodOverride() {
        rewriteRun(
          java("..."),
          java("...",
            spec -> spec.afterRecipe(cu -> {
                assertThat(...);
            })
          )
        );
    }
```

File: rewrite-java/src/main/java/org/openrewrite/java/tree/TypeUtils.java [hunk 1]
```diff
@@ -526,6 +532,10 @@ public static JavaType unknownIfNull(@Nullable JavaType t) {
         return t;
     }
 
+    public static boolean isDefaultAccessLevel(final Set<Flag> flags) {
```
Comment (timtebeek) [inline rewrite-java/src/main/java/org/openrewrite/java/tree/TypeUtils.java:535]:
```suggestion
    private static boolean isDefaultAccessLevel(final Set<Flag> flags) {
```

**Review by timtebeek — APPROVED** [review]
Thanks a lot for this fix! I've reduced the visibility on the new method you added, just to limit the public API we'd need to support going forward.

**Comment by timtebeek** [comment]
Thanks for getting this started @joteri ! I've added a test and pushed a small change such that there's less negations in the code. Hope you agree!

**Pull request merged**